cylc/rose/utilities.py

```diff
@@ -465,6 +465,7 @@ def merge_opts(config, opt_conf_keys):
         Merge all three options.
         >>> merge_opts(conf, 'gimmel')
         'aleph bet gimmel'
+        >>> os.environ['ROSE_SUITE_OPT_CONF_KEYS'] = ''
```

Comment:
I would move this bit with the env vars out into a unit test where you can monkeypatch, rather than mess with the environment at all.

Or, it turns out you can use the monkeypatch context manager
```python-console
        Merge options from defines and environment.
        >>> from pytest import MonkeyPatch
        >>> with MonkeyPatch.context() as mp:
        ...     mp.setenv('ROSE_SUITE_OPT_CONF_KEYS', 'bet')
        ...     merge_opts(conf, '')
        'aleph bet'

        Merge all three options.
        >>> with MonkeyPatch.context() as mp:
        ...     mp.setenv('ROSE_SUITE_OPT_CONF_KEYS', 'bet')
        ...     merge_opts(conf, 'gimmel')
        'aleph bet gimmel'
```

Reply:
I really want to have the example in the code, so I'll try the monkeypatching. :)